index.bs

```diff
@@ -2518,6 +2542,22 @@ interface WebTransportWriter : WritableStreamDefaultWriter {
       1. If |p| was rejected with reason |r|, then return [=a promise rejected with=] |r|.
       1. Return undefined.
 
+: <dfn for="WebTransportWriter" method>commit()</dfn>
+:: The {{commit}} method will update the {{WebTransportSendStream/[[CommittedOffset]]}} of a stream
+   to match the number of bytes written to that stream ({{WebTransportSendStream/[[BytesWritten]]}}).
+   This ensures that those bytes will be delivered to a peer reliably,
+   even after writing is [=aborted=], causing the stream to be [=stream/reset=].
+   This uses the mechanism described in [[RELIABLE-RESET]].
+
+   Note: This does not guarantee delivery in the event that a connection fails,
+   only when a stream is [=stream/reset=].
+
+   When {{commit}} is called for |stream|, the user agent MUST run the following steps:
+   1. Let |transport| be |stream|.{{WebTransportSendStream/[[Transport]]}}.
+   1. [=Queue a network task=] with |transport|
+      to set |stream|.{{WebTransportSendStream/[[CommittedOffset]]}}
+      to the value of |stream|.{{WebTransportSendStream/[[BytesWritten]]}}.
```

Comment:
Queuing a task doesn't seem needed and only adds uncertainty (values may have updated by then).
```suggestion
   1. Set |stream|.{{WebTransportSendStream/[[CommittedOffset]]}}
      to the value of |stream|.{{WebTransportSendStream/[[BytesWritten]]}}.
```

Reply:
Can we do that?  Maybe the web platform is able to play fast and loose with thread safety, but I figured that this is a case where could afford to have a single owner for the value.  If we pretend that all values are atomic, then maybe that works.  I'll leave this to you to resolve, since I don't care that much.

Reply:
For browser-initiated actions, queuing a task is essential to get on the JS event loop, but this being a method call we're already on it.

Both `[[CommittedOffset]]` and `[[BytesWritten]]` are JS-thread only, the latter incremented [here](https://pr-preview.s3.amazonaws.com/martinthomson/webtransport/pull/667.html#ref-for-dom-webtransportsendstream-byteswritten-slot%E2%91%A0) on the task that resolves the write promise, so it cannot change from under us (we're either in a different task, or more typically based on the desired `writer.write(x).then(() => writer.commit())` pattern, on the microtask queue being emptied at the tail of the same task that resolved the promise).

Reply:
Though (this reminded me) there's some prose still missing to get the the `[[CommittedOffset]]` safely from the JS thread to the in-parallel step where the reset happens. I think we need to add this to [abort a WebTransportSendStream](https://pr-preview.s3.amazonaws.com/martinthomson/webtransport/pull/667.html#webtransportsendstream-abort), and maybe `close` as well.

Reply:
Yes, it's all the same event loop. I imagine back in the days it was a single task queue as well, but someone decided this wasn't complicated enough, when input could be triaged into multiple [task sources](https://html.spec.whatwg.org/multipage/webappapis.html#task-source), by some taxonomy.